File: C++/2-Medium/00277._搜寻名人/00277._搜寻名人.cpp
#include <iostream>
#include <vector>
using namespace std;

// 假设的 knows API
bool knows(int a, int b);

class Solution {
public:
    // 寻找名人
    int findCelebrity(int n) {
        int candidate = 0;
        // 第一次遍历，找出可能的名人
        for (int i = 1; i < n; i++) {
            if (knows(candidate, i)) {
                candidate = i;
            }
        }
        
        // 第二次遍历，验证这个人是否真的是名人
        for (int i = 0; i < n; i++) {
            if (i != candidate && (knows(candidate, i) || !knows(i, candidate))) {
                return -1; // 不满足名人条件
            }
        }
        return candidate; // 返回名人编号
    }
};

// knows API 实现，此处为占位符
bool knows(int a, int b) {
    // 实际使用时，这里会有具体实现
    return true; // 假设返回值
}

int main() {
    // 示例，创建Solution实例并调用findCelebrity方法
    Solution solution;
    int n = 3; // 假设聚会中有3个人
    int celebrity = solution.findCelebrity(n);
    cout << "The celebrity is: " << celebrity << endl;
    return 0;
}

File: Python/2-Medium/00277._搜寻名人/00277._搜寻名人.py
# 假设 knows API 已经由题目提供，这里模拟实现
def knows(a: int, b: int) -> bool:
    # 这里是模拟的 knows 函数实现，具体实现应由 LeetCode 的环境提供
    # 例如，可以通过预设的二维数组来模拟人与人之间的认识关系
    pass

class Solution:
    def findCelebrity(self, n: int) -> int:
        """
        寻找名人
        
        Args:
        n: int - 人数
        
        Returns:
        int - 名人的索引，如果没有名人则返回 -1
        
        """
        # 假设第0个人是名人候选
        candidate = 0
        for i in range(1, n):
            # 如果候选人认识当前的i，则i成为新的候选人
            if knows(candidate, i):
                candidate = i
        
        # 验证候选人是否为名人
        for i in range(n):
            if i != candidate and (knows(candidate, i) or not knows(i, candidate)):
                # 如果候选人认识任何人，或者有人不认识候选人，则候选人不是名人
                return -1
        # 所有检查通过，候选人是名人
        return candidate

# 模拟 knows 函数的实现暂时省略，直接进行逻辑和实现思路的解释

# 请注意，实际的 knows API 实现和人员之间的认识关系是由 LeetCode 环境预设的，
# 这里的代码仅供算法逻辑演示之用，需要在 LeetCode 上提交验证。
